src/libraries/Microsoft.PowerFx.Interpreter/CustomFunction/ReflectionFunction.cs

```diff
@@ -156,13 +158,18 @@ internal TexlFunction GetTexlFunction()
                 };
             }
 
-            return new CustomTexlFunction(info.Name, info.RetType, info.ParamTypes)
+            return new CustomTexlFunction(info.Name, FunctionCategories.UserDefined, info.RetType, info.ParamTypes)
             {
                 _impl = (runtimeConfig, args, cancellationToken) => InvokeAsync(runtimeConfig, args, cancellationToken),
                 LamdaParamMask = info.LamdaParamMask,
             };
         }
 
+        internal string GetFunctionName()
+        {
+            return _info?.Name;

```

Comment:
>_info [](http://example.com/codeflow?start=19&length=5)

nit: To avoid  _info being null, I would recommend making it readonly, and no arg constructor can just call _info = Scan();

Reply:
update